libvirt.go

```diff
@@ -621,6 +621,23 @@ func getQEMUError(r response) error {
 	return nil
 }
 
+func (l *Libvirt) listenAndRoute() {
+	// only returns once it detects a non-temporary error related to the
+	// underlying connection
+	l.listen()
+
+	// close event streams
+	for _, ev := range l.events {
+		l.unsubscribeEvents(ev)
```

Comment:
not your doing, let's ignore for now - iirc, unsubscribeEvents() can hang forever.

Reply:
I shall keep that in mind for future